homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,247 @@
+"""Support for the WatchYourLAN service."""
+
+from homeassistant.components.sensor import (
+    RestoreSensor,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN
+from .coordinator import WatchYourLANUpdateCoordinator
+
+# Define entity descriptions for each sensor type
+ONLINE_STATUS_DESCRIPTION = SensorEntityDescription(
+    key="online_status",
+    name="Online Status",
+)
+
+IP_ADDRESS_DESCRIPTION = SensorEntityDescription(
+    key="ip_address",
+    name="IP Address",
+)
+
+MAC_ADDRESS_DESCRIPTION = SensorEntityDescription(
+    key="mac_address",
+    name="MAC Address",
+)
+
+IFACE_DESCRIPTION = SensorEntityDescription(
+    key="iface",
+    name="Network Interface",
+)
+
+
+# Utility function for setting up device info consistently
+def build_device_info(device, domain):
+    """Build device info helper function."""
+    return {
+        "identifiers": {(domain, device.get("ID", "Unknown"))},
+        "name": device.get("Name") or f"WatchYourLAN {device.get('ID', 'Unknown')}",
+        "manufacturer": device.get("Hw", "Unknown Manufacturer"),
+        "model": "WatchYourLAN Device",
+        "sw_version": device.get("Last_Seen", "Unknown"),
+    }
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the WatchYourLAN sensors."""
+    coordinator = entry.runtime_data["coordinator"]
+
+    # Define a list that can store multiple sensor types
+    entities: list[SensorEntity | CoordinatorEntity] = []
+
+    # Create device count sensor
+    entities.append(WatchYourLANDeviceCountSensor(coordinator))
+
+    # Loop over each device and create sensors for online status, IP, MAC, and Iface
+    for device in coordinator.data:
+        if isinstance(device, dict):
+            entities.append(
+                WatchYourLANSensor(coordinator, device)
+            )  # Online/Offline sensor
+            entities.append(WatchYourLANIPSensor(coordinator, device))  # IP sensor
+            entities.append(WatchYourLANMacSensor(coordinator, device))  # MAC sensor
+            entities.append(
+                WatchYourLANIfaceSensor(coordinator, device)
+            )  # Iface sensor
+
+    # Add all the entities
+    async_add_entities(entities)
+
+
+class WatchYourLANSensor(CoordinatorEntity, RestoreSensor, SensorEntity):
```

Comment:
```suggestion
class WatchYourLANSensor(CoordinatorEntity[..], RestoreSensor, SensorEntity):
```
Please add the coordinator type in the CoordinatorEntity